### PR TITLE
MRG, FIX: Corrmap check

### LINKED ISSUE #6836: mne.preprocessing.corrmap auto threshold error handling
using the threshold="auto" option in corrmap sets the lower threshold for correlation at 0.6 - if there is no correlation above that threshold, the routine dies with the not terribly informative "ValueError: x or y has zero length". Could either have a lower min threshold that's checked, or just provide feedback that there are no maps correlating above threshold.

### PR DESCRIPTION
**Pull request by larsoner**
Closes #6836.

@nugenta can you see if this gives you a more informative error now?

### CODE REVIEW

File: mne/preprocessing/ica.py [hunk 1]
```diff
@@ -2566,29 +2568,22 @@ def corrmap(icas, template, threshold="auto", label=None, ch_type="eeg",
...
-        nt, mt, s, mx = _find_max_corrs(all_maps, target, threshold)
-    else:
-        paths = [_find_max_corrs(all_maps, target, t) for t in threshold]
-        # find iteration with highest avg correlation with target
-        nt, mt, s, mx = paths[np.argmax([path[2] for path in paths])]
+    threshold = np.atleast_1d(np.array(threshold, float)).ravel()
+    threshold_err = ('No component detected using find_outliers when '
+                     'using threshold%s %s, consider using a more lenient '
+                     'threshold' % (threshold_extra, threshold))
+    if len(all_maps) == 0:
```
Comment (jona-sassenhagen) [inline mne/preprocessing/ica.py]: This is not just `if not all_maps`?

File: mne/preprocessing/ica.py [hunk 2]
```diff
@@ -2566,29 +2568,22 @@ def corrmap(icas, template, threshold="auto", label=None, ch_type="eeg",
...
-        nt, mt, s, mx = _find_max_corrs(all_maps, nt, threshold)
-    elif len(threshold) > 1:
-        paths = [_find_max_corrs(all_maps, nt, t) for t in threshold]
-        # find iteration with highest avg correlation with target
-        nt, mt, s, mx = paths[np.argmax([path[1] for path in paths])]
+    if len(all_maps) == 0 or len(nt) == 0:
+        raise RuntimeError(threshold_err)
+    paths = [_find_max_corrs(all_maps, nt, t) for t in threshold]
+    # find iteration with highest avg correlation with target
+    nt, mt, s, mx = paths[np.argmax([path[1] for path in paths])]
```
Comment (jona-sassenhagen) [inline mne/preprocessing/ica.py]: These single-character names look painful on re-read.

File: mne/preprocessing/ica.py (hunk 1 above)
Comment (larsoner) [reply mne/preprocessing/ica.py]: Yes but I find this to be more explicit about what it's checking

**Comment by larsoner** [comment]
In it goes, thanks for the look @jona-sassenhagen

**Comment by larsoner** [comment]
Backported

**Pull request merged**